alpenhorn/db/archive.py

```diff
@@ -118,3 +118,40 @@ class ArchiveFileCopyRequest(base_model):
 
     class Meta:
         indexes = ((("file", "group_to", "node_from"), False),)  # non-unique index
+
+
+class ArchiveFileImportRequest(base_model):
+    """Requests for the import of new files into a node.
+
+    Attributes
+    ----------
+    node : foreign key
+        The StorageNode on which the import should happen
+    path : string
+        The path to import.  If this is a directory and
+        recurse is True, it will be recursively scanned.
+    recurse : bool
+        If True, recursively scan path for files, if path is a directory.
+    register : bool
+        If False, only files with existing ArchiveFile records will be
+        imported.
+    completed : bool
+        Set to true when the import request has completed.
+    timestamp : datetime
+        The UTC time when the request was made.
+    """
+
+    node = pw.ForeignKeyField(StorageNode, backref="import_requests")
+    path = pw.CharField(max_length=1024)
+    recurse = pw.BooleanField(default=False)
+    register = pw.BooleanField(default=False)
+    completed = pw.BooleanField(default=False)
+    timestamp = pw.DateTimeField(default=pw.utcnow, null=True)
+
+    def complete(self) -> None:
+        """Set this request to complete in the database."""
+
+        # If we're already complete, this function does nothing
+        if not self.completed:
+            self.completed = 1
```

Comment:
why `1` instead of `True`?

Reply:
It's the same, though maybe it would be clearer to use True.  SQL doesn't have a boolean type, so when we make bool fields what we're actually making is a 1-byte int.  Peewee converts True/False to 1/0 when writing to the database.

I've been preferring 0/1 instead over False/True recently because of flake8 always complaining about the, e.g., `ArchiveFileCopyRequest == False` constructs in `where` clauses.  (Flake8 wants `is False` in this case, but that doesn't work for peewee because `is` can't be overloaded like `==` can.)

Maybe for assignment it would be better to use `True`, especially in this function where we see the `BooleanField` just above it.

Reply:
Makes sense. I'm not bothered either way, so I think it's fine to just stick with `1`

Reply:
I've changed it to `True`.  It's probably less confusing this way.